msg_helpers.go

```diff
@@ -659,6 +659,47 @@ func packDataNsec(bitmap []uint16, msg []byte, off int) (int, error) {
 	return off, nil
 }
 
+// BROKEN, TODO
+func unpackDataSvc(msg []byte, off int) ([]SvcKeyValue, int, error) {
+	var xs []SvcKeyValue
+	for off < len(msg) {
+		code, off, err := unpackUint16(msg, off)
+		length, off, err := unpackUint16(msg, off)
+		if err != nil || off+int(length) > len(msg) {
+			return nil, len(msg), &Error{err: "overflow unpacking svc"}
+		}
+		val := make([]byte, length)
+		// TODO copy assumes all keys are unrecognized
+		copy(val, msg[off:off+int(length)])
+
+		xs = append(xs, SvcKeyValue{SvcParamKey: code,
+			SvcParamValue: string(val)})
+	}
+
+	return xs, off, nil
+}
+
+func packDataSvc(options []SvcKeyValue, msg []byte, off int) (int, error) {
+	for _, el := range options {
+		off, err := packUint16(el.SvcParamKey, msg, off)
```

Comment:
I know that this is probably going to change anyway, but note that the `off` variable here shadows the `off` argument variable, so this function always ends up returning the same value that was passed in (effectively the svc fields are not encoded in the output message).